### PR TITLE
RavenDB-21088 fix

### PR DESCRIPTION
**Pull request by efratshenhar**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-21088

### Additional description

No need for read & write

### Type of change

- [x] Bug fix
- [ ] Regression bug fix
- [ ] Optimization
- [ ] New feature

### How risky is the change?

- [x] Low 
- [ ] Moderate 
- [ ] High
- [ ] Not relevant

### Backward compatibility

- [x] Non breaking change
- [ ] Ensured. Please explain how has it been implemented?
- [ ] Breaking change
- [ ] Not relevant

### Is it platform specific issue?

- [ ] Yes. Please list the affected platforms.
- [x] No

### Documentation update

- [ ] This change requires a documentation update. Please mark the issue on YouTrack using `Documentation Required` tag.
- [x] No documentation update is needed 

### Testing by Contributor

- [x] Tests have been added that prove the fix is effective or that the feature works
- [ ] Internal classes added to the test class (e.g. entity or index definition classes) have the lowest possible access modifier (preferable `private`) 
- [ ] It has been verified by manual testing

### Testing by RavenDB QA team

- [ ] This change requires a special QA testing due to possible performance or resources usage implications (CPU, memory, IO). Please mark the issue on YouTrack using `QA Required` tag.
- [x] No special testing by RavenDB QA team is needed

### Is there any existing behavior change of other features due to this change?

- [ ] Yes. Please list the affected features/subsystems and provide appropriate explanation
- [x] No

### UI work

- [ ] It requires further work in the Studio. Please mark the issue on YouTrack using `Studio Required` tag.
- [x] No UI work is needed

### CODE REVIEW

**Comment by ppekrol** [comment]
Don't you have t in 5.4 as well?

**Comment by efratshenhar** [comment]
Yes. Changed the PR to V5.4

File: src/Raven.Server/Documents/Handlers/Debugging/StorageHandler.cs [hunk 1]
```diff
@@ -380,14 +380,17 @@ await using (var writer = new AsyncBlittableJsonTextWriter(context, ResponseBody
         {
             bool first = true;
 
-            writer.WriteStartObject();
+            if (envs.Count() > 1)
```
Comment (ppekrol) [inline src/Raven.Server/Documents/Handlers/Debugging/StorageHandler.cs]: It means then that WriteReport should be per env, and code inside this `if` should be handled outside

Comment (efratshenhar) [reply src/Raven.Server/Documents/Handlers/Debugging/StorageHandler.cs]: Done

File: src/Raven.Server/Documents/Handlers/Debugging/StorageHandler.cs [hunk 2]
```diff
@@ -345,49 +345,46 @@ public async Task ScratchBufferPoolInfoReport()
...
-            else
-            {
-                envs = Database.GetAllStoragesEnvironment();
-            }
 
-            if (envs == null)
-            {
-                HttpContext.Response.StatusCode = (int)HttpStatusCode.NotFound;
-                return;
+                using (ContextPool.AllocateOperationContext(out DocumentsOperationContext context))
```
Comment (ppekrol) [inline src/Raven.Server/Documents/Handlers/Debugging/StorageHandler.cs:355]: https://github.com/ravendb/ravendb/pull/18055 returns 404 in this case, also old code. What it should be?

Comment (efratshenhar) [reply src/Raven.Server/Documents/Handlers/Debugging/StorageHandler.cs:355]: we should return 
https://github.com/ravendb/ravendb/pull/18071

Comment (efratshenhar) [reply src/Raven.Server/Documents/Handlers/Debugging/StorageHandler.cs:355]: for v6.0 https://github.com/ravendb/ravendb/pull/18072

**Pull request merged**